src/sdltiles.cpp

```diff
@@ -2041,7 +2041,18 @@ void musicFinished() {
     }
 
     // Load the next file to play.
-    current_playlist_at++;
+    if( list.shuffle ) {
+        size_t next_playlist_at = current_playlist_at;
+        // We'll retry until we get a random number that isn't our current one.
+        while( next_playlist_at == current_playlist_at ) {
```

Comment:
Going to be fun with a playlist that contains only one entry (fun as in "infinite loop").

Btw. in my opinion, if you add a restriction, it's not random anymore.